src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingToAndFromManyAddressesSpecification_Steps.cs

```diff
@@ -0,0 +1,178 @@
+        private int CoinBaseMaturity;
+
+        // Output helper allows test step names to be logged.
+        public SendingToAndFromManyAddressesSpecification(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+        protected override void BeforeTest()
+        {
+            this.sharedSteps = new SharedSteps();
+            this.nodeGroupBuilder = new NodeGroupBuilder();
+        }
+
+        protected override void AfterTest()
+        {
+            this.nodeGroupBuilder.Dispose();
+        }
+
+        private void two_connected_nodes()
+        {
+            this.nodes = this.nodeGroupBuilder
+                .StratisPowNode(NodeOne).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .StratisPowNode(NodeTwo).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .WithConnections()
+                    .Connect(NodeOne, NodeTwo)
+                    .AndNoMoreConnections()
+                .Build();
+        }
+
+        private void node1_sends_funds_to_node2_TO_fifty_addresses()
+        {
+            this.CoinBaseMaturity = (int)this.nodes[NodeOne].FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            this.sharedSteps.MineBlocks(this.CoinBaseMaturity + 50
+                , this.nodes[NodeOne]
+                , WalletAccountName
+                , WalletName
+                , WalletPassword);
+
+            this.nodes[NodeOne].CreateRPCClient().AddNode(this.nodes[NodeTwo].Endpoint, true);
+
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[NodeTwo], this.nodes[NodeOne]);
+
+            IEnumerable<HdAddress> nodeTwoAddresses = this.nodes[NodeTwo]
+                .FullNode.WalletManager()
+                .GetUnusedAddresses(new WalletAccountReference(WalletName, WalletAccountName), 50);
+
+            List<Recipient> nodeTwoRecipients = nodeTwoAddresses.Select(address => new Recipient
+            {
+                ScriptPubKey = address.ScriptPubKey,
+                Amount = Money.COIN
+            }).ToList();
+
+            this.transactionBuildContext = SharedSteps.CreateTransactionBuildContext(WalletName
+                , WalletAccountName
+                , WalletPassword
+                , nodeTwoRecipients
+                , FeeType.Medium
+                , 101);
+
+            Transaction transaction = this.nodes[NodeOne].FullNode.WalletTransactionHandler().
+                BuildTransaction(this.transactionBuildContext);
+
+            transaction.Outputs.Count.Should().Be(51);
+
+            this.nodes[NodeOne].FullNode.NodeService<WalletController>().
+                SendTransaction(new SendTransactionRequest(transaction.ToHex()));
+
+            TestHelper.WaitLoop(() => this.nodes[NodeTwo].CreateRPCClient().GetRawMempool().Length > 0);
+            TestHelper.WaitLoop(() => this.nodes[NodeTwo].FullNode.WalletManager().GetSpendableTransactionsInWallet(WalletName).Any());
+        }
```

Comment:
so you are checking it got to the mempool of the second node? that feels like it should be part of the Then and not the When?

Reply:
TBH, you could argue part of the Send step, but perhaps more so if you were checking the mem pool on the sending node instead of receiving here.